### ISSUE TITLE
[SCREENREADER]: Text and inputs added dynamically should be announced by screen readers

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#316]
## Issue
New inputs or conditional text added to the DOM based on user interaction with radio buttons, checkboxes, or select menus need to be announced to screen readers so users have context that new information is available.

## Steps to Reproduce
1. Review a form with radio buttons or checkboxes that conditionally expose additional content. This might be text or an input field.
2. Verify the new information or fields are not read out by the screen reader

## Related Issues
* https://github.com/usds/us-forms-system/issues/279

## Acceptance Criteria
The new information should be read out by the screen reader. We want users to know the form or view has changed, and additional contextual information is being presented. Screenshots below show a few instances of this pattern.

## Possible Fix
Works in Progress:
* https://github.com/usds/us-forms-system/issues/279#issuecomment-424064503

---

## Screenshots
<img width="546" alt="screen shot 2018-09-20 at 10 26 48 pm" src="https://user-images.githubusercontent.com/934879/45859205-ac2f7b80-bd26-11e8-95c3-a7b758cc91ee.png">

**Comment by peterhill-oddball** [comment]
Migrated to VA.gov ticket [1072](https://github.com/department-of-veterans-affairs/va.gov-team/issues/1072).

**Comment by 1Copenut** [comment]
@caw310 I'd like to discuss this one at our next Design System Council. It's an older issue that has come up again lately because of work on the wizards, where users are answering questions via radio buttons and are prompted with the next conditional piece of information.

**Comment by Mottie** [comment]
The `widgetProps` `ui:options` addition has been merged into the code base. Hopefully, this will allow screenreader users to have an improved experience.

**Comment by joshkimux** [comment]
The more this issue emerges, the more I believe [gov.uk's "check if a service is suitable" pattern](https://design-system.service.gov.uk/patterns/check-a-service-is-suitable/) should be considered as a replacement to the wizard to reduce cognitive load and accessibility issues.

**Comment by SarahKay8** [comment]
The dynamic wizard has been retired and @joshkimux  doesn't believe there are any more instances where text will appear dynamically within the radio group itself. There are instances where new information or alerts will appear below and outside the radio group however, but he can't remember exactly where those instances are located. He thinks one may exist in the 526 flow, but it should have been fixed. @mottie will know more about this.

**Comment by Mottie** [comment]
I do recall some pages within the 526 flow that still have an expanding radio group. I'll have to find and share the pages when I get a chance.

**Comment by joshkimux** [comment]
@k80bowman great catch! It's similar, but not the same. The issue in that ticket is using `aria-describedby` to _describe_ the radio option that has additional child content.

The issue in this ticket (I believe) is suggesting to use `aria-live` to _announce_ additional content.

That being said, the more I think about it, the more I wonder if either of these are still necessary and if a deeper discussion should be held around the designed purpose of this interaction (and if it's necessary).

Ideally, we wouldn't have to rely on both as:
- `aria-describedby` is not the most reliable way of announcing any important additional content
- `aria-live` (specifically polite) is better than `aria-describedby` in this instance as it's guaranteed to announce and won't be masked by more advanced assistive tech settings, but it can still be easily missed if they don't hear it the first time

The best alternative would be to provide any dynamic updates on the following page using a one thing per page pattern.

That being said, there still may be unique scenarios where either might be needed to in situations where we can't really do one thing per page (but I can't really come up with immediate examples right off the top of my head). 

tl;dr: a discussion might be helpful to figure out a better way forward for these two tickets 🙇

**Comment by k80bowman** [comment]
@joshkimux @caw310 would the DSC be the proper venue for that discussion?

**Comment by joshkimux** [comment]
@k80bowman it's very a11y specific, so I wonder if it might be worth a smaller huddle first?

**Comment by k80bowman** [comment]
@joshkimux a smaller huddle would be great.

**Comment by k80bowman** [comment]
I set up a meeting for June 20th to discuss this. Josh has said he would like us to explore:
- Do we need to support this pattern in the future? (with check eligibility being a potential experimental design pattern)
- If we do, what rules do we want to set based on gov.uk's research on the same topic? (they did research that suggested it should only be used for very simple, short, and predictable interactions)

**Comment by k80bowman** [comment]
Met with @humancompanion-usds and @davidakennedy this morning. Moving this ticket back to Current Sprint until work begins.

For the short-term, @davidakennedy is going to provide some guidance about handling this issue in the now-deprecated wizard pattern. Once his guidance is added to the [wizard pattern documentation](https://design.va.gov/patterns/wizards), we should be able to close this ticket.

Long-term, resolving this issue is tied up with our efforts to replace the wizard pattern with something more accessible. @humancompanion-usds is working now to find a team to test a replacement pattern.

**Comment by humancompanion-usds** [comment]
@davidakennedy - I just now deployed your doc in the Wizard pattern as well as shipped the Sub-task pattern that replaces it:

https://design.va.gov/patterns/wizards#retrofit-this-pattern
https://design.va.gov/patterns/forms/sub-task

Let me know if you have any comments on either and we'll file new issues.

**Comment by davidakennedy** [comment]
@humancompanion-usds Nothing to add. Everything looks good, and I'm hoping this helps teams!

**Comment by joshkimux** [comment]
Seeing a 3 year long ticket finally get closed is a wonderful thing. Huge thank you to everyone who worked on this, y'all are awesome 🥇
